packages/wonder-blocks-popover/components/popover-content.js

```diff
@@ -122,29 +135,75 @@ export default class PopoverContent extends React.Component<Props> {
         } = this.props;
 
         return (
-            <PopoverContentCore
-                color={emphasized ? "blue" : "light"}
-                closeButtonLabel={closeButtonLabel}
-                closeButtonVisible={closeButtonVisible}
-                onClose={onClose}
-                style={style}
-                testId={testId}
-            >
-                <View style={!!icon && styles.withIcon}>
-                    {image && <StyledImage style={styles.image} src={image} />}
-
-                    {icon && <StyledImage style={styles.icon} src={icon} />}
-
-                    <View style={styles.text}>
-                        <HeadingSmall style={styles.title}>
-                            {title}
-                        </HeadingSmall>
-                        <Body>{content}</Body>
-                    </View>
-                </View>
-
-                {actions && <View style={styles.actions}>{actions}</View>}
-            </PopoverContentCore>
+            <PopoverContext.Consumer>
+                {({close, placement}) => {
+                    if (close && onClose) {
+                        throw new Error(
+                            "You've specified 'onClose' on the content when using Popover. Please specify 'onClose' on the Popover instead",
+                        );
+                    }
+
+                    // illustration popover can't be placed horizontally
+                    if (
+                        image &&
+                        (placement === "left" || placement === "right")
+                    ) {
+                        throw new Error(
+                            "'image' can only be vertically placed. You can fix this by either changing `placement` to `top` or `bottom` or removing the `image` prop inside `content`.",
+                        );
+                    }
```

Comment:
I wonder if we should encapsulate these error checks in a `validateProps` call or something to make this inline function a little easier to follow.